syncer/syncer.go

```diff
@@ -856,7 +858,16 @@ func (s *Syncer) flushCheckPoints() error {
 		shardMetaSQLs, shardMetaArgs = s.sgk.PrepareFlushSQLs(exceptTableIDs)
 		s.tctx.L().Info("prepare flush sqls", zap.Strings("shard meta sqls", shardMetaSQLs), zap.Reflect("shard meta arguments", shardMetaArgs))
 	}
-	err := s.checkpoint.FlushPointsExcept(exceptTables, shardMetaSQLs, shardMetaArgs)
+
+	// when canceling (stop-task/pause-task), we still need to flush the checkpoint.
+	timeout, err := time.ParseDuration(maxDMLConnectionTimeout)
```

Comment:
we can define a new variable with `time.Duration` type, and then don't need parse every time.

Reply:
addressed in https://github.com/pingcap/dm/pull/377/commits/f59a6aace6dfb8ad49ea64d149674d56a1cb3ff9.